src/ripple/protocol/impl/SecretKey.cpp

```diff
@@ -178,16 +147,27 @@ randomSecretKey()
 
 // VFALCO TODO Rewrite all this without using OpenSSL
 //             or calling into GenerateDetermisticKey
-
 SecretKey
-generateSecretKey (Seed const& seed)
+generateSecretKey (KeyType type, Seed const& seed)
 {
-    uint128 ps;
-    std::memcpy(ps.data(),
-        seed.data(), seed.size());
-    auto const upk =
-        generateRootDeterministicPrivateKey(ps);
-    return SecretKey(Slice{ upk.data(), upk.size() });
+    if (type == KeyType::ed25519)
+    {
+        auto const key = sha512Half_s(Slice(
+            seed.data(), seed.size()));
+        return SecretKey(Slice{ key.data(), key.size() });
+    }
+
+    if (type == KeyType::secp256k1)
+    {
+        uint128 ps;
+        std::memcpy(ps.data(),
+            seed.data(), seed.size());
+        auto const upk =
+            generateRootDeterministicPrivateKey(ps);
+        return SecretKey(Slice{ upk.data(), upk.size() });
+    }
```

Comment:
Sorry, "methods" wasn't the right word. What I don't get is why we need two separate blocks with different methodologies for ed25519 and secp256k1, when we had the secp code before for all cases.